pebblo/app/service/local_ui/loader_apps.py

```diff
@@ -0,0 +1,490 @@
+        try:
+            self.db = SQLiteClient()
+
+            # create session
+            self.db.create_session()
+
+            _, ai_loader_apps = self.db.query(table_obj=AiDataLoaderTable)
+            logger.debug(f"LoaderAppDetailsObject; {ai_loader_apps}")
+
+            # Preparing all loader apps
+            all_loader_apps: list = []
+            for loader_app in ai_loader_apps:
+                app_data = loader_app.data
+                logger.debug(f"LoaderEachApp: {app_data}")
+                if app_data.get("docEntities") not in [None, {}] or app_data.get(
+                    "docTopics"
+                ) not in [None, {}]:
+                    self.loader_apps_at_risk += 1
+                    loader_app = self.get_findings_for_loader_app(app_data)
+                    all_loader_apps.append(loader_app)
+
+            # TODO: Sort loader apps
+            # sorted_loader_apps = self._sort_loader_apps(all_loader_apps)
+
+            logger.debug("[Dashboard]: Preparing loader app response object")
+            loader_response = LoaderAppModel(
+                applicationsAtRiskCount=self.loader_apps_at_risk,
+                findingsCount=self.loader_findings,
+                documentsWithFindingsCount=self.loader_files_findings,
+                dataSourceCount=self.loader_data_source,
+                appList=all_loader_apps,
+                findings=self.loader_findings_list,
+                documentsWithFindings=self.loader_document_with_findings_list,
+                dataSource=self.loader_data_source_list,
+            )
+            logger.debug(f"LoaderAppResponse: {loader_response.dict()}")
+
+        except Exception as ex:
+            logger.error(f"[Dashboard]: Error in all loader app listing. Error:{ex}")
+            # Getting error, Rollback everything we did in this run.
+            self.db.session.rollback()
+
+        else:
+            # Commit will only happen when everything went well.
+            message = "All loader app response prepared successfully"
+            logger.debug(message)
+            self.db.session.commit()
+            return loader_response.dict()
+        finally:
+            logger.debug("Closing database session for preparing all loader apps")
+            # Closing the session
+            self.db.session.close()
+
+    def _pdf_writer(self, file_path, data):
+        report_obj = Reports()
+        report_format = CacheDir.FORMAT.value
+        renderer = CacheDir.RENDERER.value
+
+        full_file_path = get_full_path(file_path)
+
+        # Create parent directories if needed
+        dir_path = path.dirname(full_file_path)
+        makedirs(dir_path, exist_ok=True)
+
+        status, result = report_obj.generate_report(
+            data=data,
+            output_path=full_file_path,
+            format_string=report_format,
+            renderer=renderer,
+        )
+
+        if not status:
+            logger.error(f"PDF report is not generated. {result}")
```

Comment:
do we need to log result ?

Reply:
Yes, we print this statement on console and in log file.